res/controllers/M-Audio-Xponent-Advanced-scripts.js

```diff
@@ -0,0 +1,842 @@
+    "pfl": 0x14,
+    "scratch": 0x15,
+    "punchIn": 0x07,
+    "sync": 0x02,
+    "button_parameter1": 0x08,
+    "button_parameter2": 0x09,
+    "button_parameter3": 0x0A,
+    "gain": 0x0B,
+    "shift": 0x2C
+};
+
+MaudioXponent.binleds = {
+    8: "button_parameter1",
+    9: "button_parameter2",
+    10: "button_parameter3",
+    16: (MaudioXponent.config.nudgeButtonMode ? "rate_temp_up" : "rate_temp_down"),
+    17: (MaudioXponent.config.nudgeButtonMode ? "rate_temp_down" : "rate_temp_up"),
+    18: "keylock",
+    19: "reverse",
+    20: "pfl",
+    33: "back",
+    34: "fwd",
+    36: "play"
+};
+
+// ----------   Functions    ----------
+MaudioXponent.logParams = function(functionName, a, b, c, d, e, f) {
+    print("***");
+    print("*** " + functionName + ": a="+ a + ", b=" + b + ", c=" + c + ", d=" + d + ", e=" + e + ", f=" + f);
+    print("***");
+};
+
+MaudioXponent.init = function (id) {
+    MaudioXponent.initDecks();
+    MaudioXponent.initLights();
+    MaudioXponent.syncLights();
+
+    //MaudioXponent.probeLights();
+};
+
+MaudioXponent.initDecks = function() {
+    // Because vuMeter values change rapidly, we're pre-digesting the parameters for this function
+    engine.connectControl("[Master]", "VuMeterL", function(value) { MaudioXponent.vuMeter(0, 0, 0, value); });
+    engine.connectControl("[Master]", "VuMeterR", function(value) { MaudioXponent.vuMeter(0, 0, 1, value); });
+    engine.connectControl("[Channel1]", "VuMeterL", function(value) { MaudioXponent.vuMeter(1, MaudioXponent.bankA, 0, value); });
+    engine.connectControl("[Channel2]", "VuMeterR", function(value) { MaudioXponent.vuMeter(1, MaudioXponent.bankA, 1, value); });
+    engine.connectControl("[Channel3]", "VuMeterL", function(value) { MaudioXponent.vuMeter(1, MaudioXponent.bankB, 0, value); });
+    engine.connectControl("[Channel4]", "VuMeterR", function(value) { MaudioXponent.vuMeter(1, MaudioXponent.bankB, 1, value); });
+
+    for (channel = 1; channel <= MaudioXponent.maxDecks; channel++) {
+        var group = "[Channel" + (channel) + "]";
+        var deck = MaudioXponent.decks[channel - 1];
+        deck.id = channel;
+        deck.beatState = false;
+        deck.group = group;
+        deck.progressMeterStatusByte = deck.isBankA ? 0xB3 : 0xB8;
+        deck.progressMeterSecondByte = deck.isLeft ? 0x14 : 0x15;
+        deck.vuMeterStatusByte = deck.isBankA ? 0xB3 : 0xB8;
+        deck.vuMeterSecondByte = deck.isLeft ? 0x12 : 0x13;
+        deck.scratchEnabled = false;
+        deck.scratching = false;
+        deck.shift = false;
+        deck.warnAt = 0;
+        deck.filterLow = 1;
+        deck.filterMid = 1;
+        deck.filterHigh = 1;
+        deck.pregain = 1;
+
+        engine.connectControl(group, "play_indicator", "MaudioXponent.onPlay");
+        engine.connectControl(group, "playposition", "MaudioXponent.onPlayPositionChange");
+        engine.connectControl(group, "duration", "MaudioXponent.onTrackLoaded");
```

Comment:
I saw that. It's just not available to 2.0, which I still need to use as a "daily driver" until 2.1 is stable.